src/ui/atoms/chip/index.tsx

```diff
@@ -26,91 +26,128 @@ import { Variant } from 'lib/types';
  *
  */
 
-interface ChipProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
-  children?: never;
+interface ChipProps {
+  action?: React.ReactNode;
+  children?: string;
   className?: string;
-  icon?: React.ReactNode;
-  onClick?: React.MouseEventHandler<HTMLButtonElement>;
-  text: string;
+  disabled?: boolean;
+  onActionClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
 }
 
 const ChipBase: React.FC<ChipProps & Variant> = ({
+  action,
+  children,
   className,
-  icon,
+  disabled,
+  onActionClick,
   onClick,
-  text,
   variant = 'default',
-  ...p
-}) => (
-  <>
-    <button
-      type="button"
-      onClick={onClick}
-      className={className}
-      data-variant={variant}
-      {...p}
-    >
-      {text}
-      <span data-icon="right-icon">{icon}</span>
-    </button>
-  </>
-);
+}) => {
+  const divRole = onClick ? 'button' : 'div';
+  const divTabIndex = onClick ? 1 : 0 ;
+  return(
```

Comment:
```suggestion
  return (
```

похоже, что тут потерялось форматирование